### PR TITLE
 Extract specification snippets as standalone JSON files (latest)

### PR DESCRIPTION
**Pull request by sbesson**
Follow-up of #98, this ports the additions made to the 0.4 specification i.e. the extraction of the JSON snippets and new DOcuments conventions paragraph to the `latest` specification.

Additionally, 0765b4d4aa00509e0263887a425f38f7a930028e and  ff936b894f9047508849bee8072b6388d0ac935a proposes to add the rest of the validation infrastructure i.e. the JSON schemas and the validation tests run as part of the CI to the `latest` subfolder.
From my perspective, the secondary benefit of adding this infrastructure is to make the JSON examples and schemas truly first-class citizens of upcoming NGFF proposals made against `latest`. The primary downside is that it might introduce some friction for instance if multiple features are under development. Interesting to hear what people who proposed the next set of breaking changes of the specification feel about this approach especially @bogovicj (#94) and @joshmoore  (#104).

### CODE REVIEW

**Comment by bogovicj** [comment]
Nice @sbesson !

>  make the JSON examples and schemas truly first-class citizens of upcoming NGFF proposals 

My first impression is that this is a really idea. Better to have to think about existing examples when making changes than to forget about them!  I'll look more carefully next week

File: latest/schemas/strict_image.schema [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "$id": "https://ngff.openmicroscopy.org/0.4/schemas/strict_image.schema",
+  "allOf": [
+    {
+      "$ref": "https://ngff.openmicroscopy.org/0.4/schemas/image.schema"
```
Comment (will-moore) [inline latest/schemas/strict_image.schema:5]: Do these IDs get updated directly to `/0.5/` at this point? (don't need to use `/latest/`, right?)
NB: I notice that https://ngff.openmicroscopy.org/0.4/schemas/image.schema is 404, so this schemas won't work without the custom resolver used in the tests.

Comment (sbesson) [reply latest/schemas/strict_image.schema:5]: No you are correct,  I didn't do the update work as I was waiting to get a consensus about the value of adding these scneas.

A few comments:
- definitely need to replace `0.4`. I would not use `0.5`  as that would communicate a released schema and that has the potential of creating confusion. `latest` would be an option or maybe something that explicitly says this is a development schema e.g. `0.5dev`?
- the individual `version` elements in the JSON snippets/examples will likely need to be updated to follow the same convention
- [the release process](https://github.com/ome/ngff/blob/main/README.md#release-process) will need to be amended to include these steps

**Comment by joshmoore** [comment]
https://api.csswg.org/bikeshed/?url=https://raw.githubusercontent.com/sbesson/ngff/json_snippets_latest/latest/index.bs continues to look good. There are still a few `0.4` matches in the document. For those in the index.bs itself, adding:

```
Text Macro: NGFFVERSION 0.5-dev
```

to the header will lead to replacement. This unfortunately does *not* work for the `<pre class=include-code>` block. (Only `class=include` which doesn't do any formatting.)

**Comment by sbesson** [comment]
Created a text macro to manage the replacements (all the other usages of 0.4 are legit and refer either to the stable version and/or to the version history). Also replaced the versions in the JSON snippets manually. My intent once this is merged is to extract all included code blocks into standalone JSON files so that we can validate them via JSON schemas and include them in the spec.

File: latest/examples/valid_strict/multiscales_example.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    "path": "2",
+                    "coordinateTransformations": [{
+                        // the voxel size for the third scale level (downscaled by a factor of 4 -> 2 micrometer)
+                        "type": "scale",
+                        "scale": [1.0, 1.0, 2.0, 2.0, 2.0]
+                    }]
+                }
+            ],
+            "coordinateTransformations": [{
+                // the time unit (0.1 milliseconds), which is the same for each scale level
```
Comment (will-moore) [inline latest/examples/valid_strict/multiscales_example.json:40]: Are these comments valid JSON? Do they cause any issues? (I see they are the same as this under 0.4/)

Comment (joshmoore) [reply latest/examples/valid_strict/multiscales_example.json:40]: from https://github.com/ome/ngff/pull/98#issuecomment-1070598704

> a new Document conventions section is added to the specification which ... clarifies the explanation scope of the comments in the JSON snippets

**Comment by joshmoore** [comment]
Any objections to moving forward with this?

**Comment by will-moore** [comment]
GFI 👍

**Comment by bogovicj** [comment]
looks good to me

**Pull request merged**